cli/azd/extensions/microsoft.azd.extensions/internal/helpers.go

```diff
@@ -313,3 +315,67 @@ func GetFileNameWithoutExt(filePath string) string {
 	// Remove the extension
 	return strings.TrimSuffix(fileName, filepath.Ext(fileName))
 }
+
+// HasLocalRegistry checks if a local extension source registry exists
+func HasLocalRegistry() (bool, error) {
+	cmdBytes, err := exec.Command("azd", "ext", "source", "list", "-o", "json").Output()
+	if err != nil {
+		return false, fmt.Errorf("failed to execute command: %w", err)
+	}
+
+	var extensionSources []any
+	if err := json.Unmarshal(cmdBytes, &extensionSources); err != nil {
+		return false, fmt.Errorf("failed to unmarshal command output: %w", err)
+	}
+
+	for _, source := range extensionSources {
+		extensionSource, ok := source.(map[string]any)
+		if ok {
+			if extensionSource["name"] == "local" && extensionSource["type"] == "file" {
+				return true, nil
+			}
+		}
+	}
+
+	return false, nil
+}
+
+// CreateLocalRegistry creates a local extension source registry
```

Comment:
Don't have context here.. but this seems a little odd..?

As an external process of azd, I had to:
1. Probe `AZD_CONFIG_DIR` deeply, and stuff a semi-working file `{"registry":{}}` into the azd config directory, i.e. `~/.azd/registry.json`
2. With that knowledge, call a first-party `azd` command now using just `registry.json` to register the file I created in step 1

Is there a happier path we can follow? I can definiltey provide suggestions given enough context.

Reply:
@wbreza what are your thoughts here?

Reply:
There isn't anything very fancy going on here.  We are performing the following:

- Get location of azd config directory
- Generate an empty `registry.json` file and write to azd config dir
- Calling azd extension commands to register this local registry

We are not:
- Manually inspecting azd configuration 
- Manually writing any global azd configuration